home/filter.go

```diff
@@ -468,7 +468,8 @@ func (f *Filtering) parseFilterContents(file io.Reader) (int, uint32, string) {
 		checksum = crc32.Update(checksum, crc32.IEEETable, []byte(line))
 
 		line = strings.TrimSpace(line)
-		if len(line) == 0 {
```

Comment:
Makes sense, but you should also ignore lines starting with `!`:
https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists

Reply:
Sure thing, just didn't see that referenced on the og issue. Not a problem.

Reply:
This is incorrect, because there's `if line[0] == '!'` check below.
And now we'll enter `if len(line) == 0 || isComment` if-statement.

What we should do is just add another check `if line[0] == '#' {` after `if line[0] == '!' {`